ZohoCRMCharts/Classes/ZCRMFunnel.swift

```diff
@@ -0,0 +1,1148 @@
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title]-[scrollView]|", options: [], metrics: nil, views:[ "title": self.titleView, "scrollView": self.scrollView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|[barChart(chartWidth)]|", options: [], metrics: ["chartWidth": chartWidth], views: ["barChart": self.barChart])
+		
+		if self.segments.count > 0 {
+			
+			let barChartHeight = self.frame.height * 0.45
+			
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title]-[rateView(conversionRateHeight)][segmentsView]|", options: [], metrics: ["conversionRateHeight": barChartHeight], views: ["title": self.titleView,"rateView": self.standardConversionRateView, "segmentsView": self.stackView1, "collectionView": self.collectionView])
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|[segmentsView(rateViewWidth)][scrollView]|", options: [], metrics: ["rateViewWidth": rateWidth], views: ["segmentsView": self.stackView1, "scrollView": self.scrollView])
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:[collectionView(collectionViewWidth)]", options: [], metrics: ["collectionViewWidth": cVWidth], views: ["collectionView": self.collectionView])
+
+			self.activate(constraints: constraints)
+			self.stackView1.layoutIfNeeded()
+			
+			constraints.removeAll()
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:|[barChart(barChartHeight)][collectionView(collectionViewHeight)]|", options: [], metrics: ["barChartHeight": barChartHeight , "collectionViewHeight": self.stackView1.frame.height], views: ["barChart": self.barChart, "collectionView": self.collectionView])
+			self.activate(constraints: constraints, true)
+			
+			self.collectionViewFlowLayout.itemSize = CGSize(width: cellWidth, height: self.stackView1.frame.height / self.rowCount.toCGFloat())
+			self.collectionViewFlowLayout.sectionInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+			self.collectionViewFlowLayout.minimumLineSpacing = 0
+			self.collectionViewFlowLayout.minimumInteritemSpacing = 0
+			
+		} else {
+			
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title]-[rateView]|", options: [], metrics: nil, views: ["title": self.titleView,"rateView": self.standardConversionRateView])
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:|[barChart(scrollView)]|", options: [], metrics: nil, views: ["barChart": self.barChart, "scrollView": self.scrollView])
+			self.activate(constraints: constraints)
+		}
+	}
+}
+
+//MARK: - Collection view delegate.
+
+extension ZCRMFunnel: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+	
+	public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+		
+		var count: Int!
```

Comment:
you can set count default value as Zero.

Reply:
No need for that.  It will be assigned with a value in the next code block.